internal/backend/telemetry/telemetry.go

```diff
@@ -59,72 +79,45 @@ func New(z *zap.Logger, cfg *Config) (*Telemetry, error) {
 	}
 
-	const schemaURL = "https://opentelemetry.io/schemas/1.1.0"
-	resourceAttrs := resource.NewWithAttributes(
-		schemaURL,
-		semconv.ServiceNameKey.String(telemetry.cfg.ServiceName),
-	)
-
 	// NOTE: do we need a better control ober batching/sending. Should we use controller?
-	meterProvider := sdk.NewMeterProvider(
+	meterProvider = sdk.NewMeterProvider(
 		sdk.WithReader(sdk.NewPeriodicReader(exporter)),
-		sdk.WithResource(resourceAttrs),
+		sdk.WithResource(ra),
 	)
 
-	otel.SetMeterProvider(meterProvider) // set global meter provider
-	return telemetry, nil
-}
+	otel.SetMeterProvider(meterProvider)
 
-func (t *Telemetry) ensureServiceName(name string) string {
-	if !strings.HasPrefix(name, t.cfg.ServiceName) {
-		name = fmt.Sprintf("%s.%s", t.cfg.ServiceName, name)
-	}
-	return name
-}
+	meter = &namedMeter{underlying: meterProvider.Meter(cfg.ServiceName), svcName: cfg.ServiceName}
 
-func (t *Telemetry) NewUpDownCounter(name string, description string) (metric.Int64UpDownCounter, error) {
-	if t == nil || !t.cfg.Enabled {
-		return noop.Int64UpDownCounter{}, nil
-	}
-	meter := otel.GetMeterProvider().Meter(t.cfg.ServiceName)
-	name = t.ensureServiceName(name)
-	metric, err := meter.Int64UpDownCounter(name, metric.WithDescription(description))
-	if err != nil {
-		t.l.Error("failed to create metric", zap.String("name", name), zap.Error(err))
-		return noop.Int64UpDownCounter{}, err
-	}
-	return metric, nil
+	return nil
 }
 
-func (t *Telemetry) NewCounter(name string, description string) (metric.Int64Counter, error) {
-	if t == nil || !t.cfg.Enabled {
-		return noop.Int64Counter{}, nil
+func setupTracing(cfg *Config, ra *resource.Resource) error {
+	if !cfg.Tracing {
+		return nil
 	}
-	meter := otel.GetMeterProvider().Meter(t.cfg.ServiceName)
-	name = t.ensureServiceName(name)
-	metric, err := meter.Int64Counter(name, metric.WithDescription(description))
-	if err != nil {
-		t.l.Error("failed to create metric", zap.String("name", name), zap.Error(err))
-		return noop.Int64Counter{}, err
-	}
-	return metric, nil
-}
 
-func (t *Telemetry) NewHistogram(name string, description string) (metric.Int64Histogram, error) {
-	if t == nil || !t.cfg.Enabled {
-		return noop.Int64Histogram{}, nil
-	}
-	meter := otel.GetMeterProvider().Meter(t.cfg.ServiceName)
-	name = t.ensureServiceName(name)
-	metric, err := meter.Int64Histogram(name, metric.WithDescription(description))
+	traceExporter, err := otlptracehttp.New(context.Background(), otlptracehttp.WithInsecure(), otlptracehttp.WithEndpoint(cfg.Endpoint))
 	if err != nil {
-		t.l.Error("failed to create metric", zap.String("name", name), zap.Error(err))
-		return noop.Int64Histogram{}, err
+		return fmt.Errorf("failed to create trace exporter: %w", err)
 	}
-	return metric, nil
+
+	bsp := sdktrace.NewBatchSpanProcessor(traceExporter)
+
+	tracerProvider = sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()), // TODO: This is going to be costly!
```

Comment:
Agree. @efiShtain any thoughts on this?

Reply:
changed to ratio sampling. defaults: dev=1, prod=0. need to experiment with values for prod.